core/sequencer/server.go

```diff
@@ -92,7 +102,34 @@ func NewServer(
 		tlog:      tlog,
 		tmap:      tmap,
 		mutations: mutations,
+		queue:     queue,
+	}
+}
+
+// RunBatch reads mutations out of the queue and calls CreateEpoch.
+func (s *Server) RunBatch(ctx context.Context, in *spb.RunBatchRequest) (*empty.Empty, error) {
+	batch, err := s.queue.ReadQueue(ctx, in.DomainId, in.MaxBatch)
+	if err != nil {
+		return nil, status.Errorf(codes.Internal, "ReadQueue(): %v", err)
 	}
+	if int32(len(batch)) < in.MinBatch {
+		return &empty.Empty{}, nil
+	}
+	if err := s.queue.DeleteMessages(ctx, in.DomainId, batch); err != nil {
+		return nil, err
+	}
+	msgs := make([]*ktpb.EntryUpdate, 0, len(batch))
+	for _, m := range batch {
+		msgs = append(msgs, &ktpb.EntryUpdate{
+			Mutation:  m.Mutation,
+			Committed: m.ExtraData,
+		})
+	}
+	_, err = s.CreateEpoch(ctx, &spb.CreateEpochRequest{
+		DomainId: in.DomainId,
+		Messages: msgs,
+	})
+	return &empty.Empty{}, err
```

Comment:
should this return `nil, err` if `CreateEpoch` emitted a non-`nil` error?

Reply:
I don't think it's super important to return nil in that case, but I can do one better:

```go
      return s.CreateEpoch(ctx, &spb.CreateEpochRequest{
                 DomainId: in.DomainId,
                 Messages: msgs,
        })
```